source/extensions/matching/input_matchers/runtime_fraction/matcher.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/matcher/matcher.h"
+#include "envoy/runtime/runtime.h"
+
+#include "source/common/common/hash.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace RuntimeFraction {
+
+class Matcher : public Envoy::Matcher::InputMatcher {
+public:
+  Matcher(Runtime::Loader& runtime,
+          envoy::config::core::v3::RuntimeFractionalPercent runtime_fraction, uint64_t seed)
+      : runtime_(runtime), runtime_fraction_(runtime_fraction), seed_(seed) {}
+  bool match(absl::optional<absl::string_view> input) override {
+    // Only match if the value is present.
+    if (!input) {
+      return false;
+    }
+
+    // Otherwise, match if feature is enabled for hash(input).
```

Comment:
Aha, I previously missed that `hash_value` is taking the place of `random_value` in featureEnabled. Could you make this comment clearer about that? "feature is enabled for hash(input)" is non-obvious that this is a pseudo-random decision.